### ISSUE TITLE
Weapon scale is overridden for female soldiers

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#921]
While it's possible to rescale a weapon's mesh in WeaponInitialized (the logical place to do so if you wanted to rescale a weapon), the mesh scale will be overridden by game's code that rescales weapons for female soldiers to 85% of their original size.

A simple change to the game's code so that mesh scale is **multiplied** by 0.85 rather than **set** to 0.85 will make it possible for mods to arbitrarily rescale weapons for soldiers and other units regardless of their gender, without any changes to vanilla functionality.

**Comment by Iridar** [comment]
This issue keeps being an exercise in rake stepping. Originally I made the code work by doing:

```
WeaponScale = WeaponScale * Mesh.Scale;
Mesh.SetScale(WeaponScale);
```
, but was made to realize that changing pawn's `WeaponScale` will affect other weapons equipped on the unit as well.
Then I moved to `Mesh.SetScale(WeaponScale * Mesh.Scale);` implementation, but turned out that causes a cascade effect of rescaling the weapon:
```
3076 [0108.13] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 2.0000 final scale: 1.7000
3358 [0143.03] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 1.7000 final scale: 1.4450
3437 [0143.37] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 1.4450 final scale: 1.2283
3625 [0143.91] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 1.2283 final scale: 1.0440
3641 [0148.26] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 1.0440 final scale: 0.8874
3658 [0148.28] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 0.8874 final scale: 0.7543
3663 [0148.29] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 0.7543 final scale: 0.6412
3790 [0148.49] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 0.6412 final scale: 0.5450
3813 [0155.73] IRITEST: EquipWeapon::WeaponScale: 0.8500 kWeapon.Mesh.Scale: 0.5450 final scale: 0.4632
```
I'm not sure how to solve this problem, because I would need to somehow get my hands on the weapon's initial mesh scale that was recorded onto the weapon's mesh in WeaponInitialized (or any other place after the weapon visualizer was created, but before it was attached to the soldier).

**Comment by MrNiceUK** [comment]
Try checking if the component has already been attached to the pawn, if so, can assume that it's mesh has already been scaled
`Mesh.Attachments.find('Component', MeshComp) != INDEX_NONE` ?

**Comment by Iridar** [comment]
> Try checking if the component has already been attached to the pawn, if so, can assume that it's mesh has already been scaled
> `Mesh.Attachments.find('Component', MeshComp) != INDEX_NONE` ?

Good idea, it works. However, it seems to be possible only if we rescale weapons only when they are attached for the first time, which is different from vanilla code path, that always rescales weapons when they are attached, even if they are already attached to the pawn.

I'm afraid changing this might have some kind of negative repercussions.

I'm fine with shelving this for now, I only started working on it because I thought this would be quick and easy (is anything ever?..). Maybe one day I'll feel like it and then I'll release a standalone mod with an MCO of XComUnitPawn, and if nobody reports any issues over entire campaigns we'll consider adding this to highlander.
